src/proto/Cosmos.proto

```diff
@@ -85,6 +85,53 @@ message Message {
         string type_prefix = 3;
     }
 
+    // wasm/MsgExecuteContract
+    message WasmExecuteContract {
+        // sender address
+        string sender_address = 1;
+
+        // token contract address
+        string contract_address = 2;
+
+        // base64-encoded payload of contract execution message (refer to below)
+        string execute_msg = 3;
+
+        // used in case you are sending native tokens along with this message
+        repeated string coin = 4;
+    }
+
+    // transfer within wasm/MsgExecuteContract
+    message WasmExecuteContractTransfer {
```

Comment:
what's the difference between `WasmExecuteContractTransfer` and `WasmExecuteContractSend`?

Reply:
Transfer is simple transfer to an address, Send is transfer with potential actions, so it works on contracts as well.
Services I tried use simple transfer (transfer from Station, withdraw from Kraken).
I will add implementation and tests for Send, nonetheless.

Reply:
This is interesting, user needs to check if the receiver is a contract?

Reply:
my question is can we just combine these two?